request_a_govuk_domain/request/models/review.py

```diff
@@ -18,91 +31,101 @@ class Review(models.Model):
+        choices=PolicyExemptionReviewChoices.choices,
     )
     policy_exemption_notes = models.TextField(
-        max_length=NOTES_MAX_LENGTH, blank=True, null=True
+        max_length=NOTES_MAX_LENGTH, validators=[MinLengthValidator(NOTES_MIN_LENGTH)]
     )
 
-    domain_name_rules = models.CharField(
-        choices=review_choices.DomainNameRulesReviewChoices.choices,
-        blank=True,
-        null=True,
-    )
+    domain_name_rules = models.CharField(choices=DomainNameRulesReviewChoices.choices)
     domain_name_rules_notes = models.TextField(
-        max_length=NOTES_MAX_LENGTH, blank=True, null=True
+        max_length=NOTES_MAX_LENGTH, validators=[MinLengthValidator(NOTES_MIN_LENGTH)]
     )
 
     registrant_senior_support = models.CharField(
-        choices=review_choices.RegistrantSeniorSupportReviewChoices.choices,
-        blank=True,
-        null=True,
+        choices=RegistrantSeniorSupportReviewChoices.choices
     )
     registrant_senior_support_notes = models.TextField(
-        max_length=NOTES_MAX_LENGTH, blank=True, null=True
+        max_length=NOTES_MAX_LENGTH, validators=[MinLengthValidator(NOTES_MIN_LENGTH)]
     )
 
-    registry_details = models.CharField(
-        choices=review_choices.RegistryDetailsReviewChoices.choices,
-        blank=True,
-        null=True,
-    )
+    registry_details = models.CharField(choices=RegistryDetailsReviewChoices.choices)
     registry_details_notes = models.TextField(
-        max_length=NOTES_MAX_LENGTH, blank=True, null=True
+        max_length=NOTES_MAX_LENGTH, validators=[MinLengthValidator(NOTES_MIN_LENGTH)]
     )
 
-    reason = models.TextField(max_length=NOTES_MAX_LENGTH, blank=True, null=True)
+    reason = models.TextField(
+        max_length=NOTES_MAX_LENGTH, validators=[MinLengthValidator(NOTES_MIN_LENGTH)]
+    )
 
     history = HistoricalRecords()
 
     def is_approvable(self) -> bool:
-        return True
+        if (
+            self.registrar_details == RegistrarDetailsReviewChoices.APPROVE
+            and self.domain_name_availability
+            == DomainNameAvailabilityReviewChoices.APPROVE
+            and self.registrant_org == RegistrantOrgReviewChoices.APPROVE
+            and self.registrant_person == RegistrantPersonReviewChoices.APPROVE
+            and (
+                not self.application.written_permission_evidence
+                or (
+                    self.application.written_permission_evidence
+                    and self.registrant_permission
+                    == RegistrantPermissionReviewChoices.APPROVE
+                )
+            )
+            and (
+                not self.application.policy_exemption_evidence
+                or (
+                    self.application.policy_exemption_evidence
+                    and self.policy_exemption == PolicyExemptionReviewChoices.APPROVE
+                )
+            )
+            and (
+                not self.application.ministerial_request_evidence
```

Comment:
I think we can take out the  self.application.ministerial_request_evidence from the or condition as it only reaches there if that is true? Same for the other two conditions we added above.

Reply:
Sorry I didn't think about it earlier.